crates/gas/src/lib.rs

```diff
@@ -197,8 +202,7 @@ impl FromStr for Gas {
 
     fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
         let raw: u64 = s.parse().map_err(GasParseError::Parse)?;
-        let gas = Gas::from_whole_units(raw).ok_or(GasParseError::Overflow)?;
```

Comment:
Same thing here, we should probably remove this implementation and come up with a custom method